src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.CpuUtilizationReader.Windows.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    internal partial class ClrThreadPool
+    {
+        private class CpuUtilizationReader
+        {
+            private struct ProcessCpuInformation
+            {
+                public ulong idleTime;
+                public ulong kernelTime;
+                public ulong userTime;
+            }
+
+            private ProcessCpuInformation _cpuInfo = new ProcessCpuInformation();
+
+            public CpuUtilizationReader()
+            {
+                GetCpuUtilization(); // Call once to initialize the usage buffer
```

Comment:
We still need to do this initial read to initialize our data structure. Without it, the first read would give us the CPU utilization since system startup I believe, which is not what we want.